### PR TITLE
Merge branch ApiTest into master

### PR DESCRIPTION
**Pull request by TroyNeubauer**

### CODE REVIEW

File: src/main/include/CobConstants.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+enum class CobKey {
```
Comment (Mee42) [inline src/main/include/CobConstants.h:3]: Cob setup looks great. How would enums serialize to a file?

File: src/main/include/subsystems/DriveTrain.h [hunk 2]
```diff
@@ -1,28 +1,22 @@
 #pragma once
 
 #include <frc2/command/SubsystemBase.h>
-#include <frc/WPILib.h>
+#include <ctre/Phoenix.h>
```
Comment (Mee42) [inline src/main/include/subsystems/DriveTrain.h:4]: Did you get this compiling?

**Review by Mee42 — APPROVED** [review]

File: src/main/include/CobConstants.h (hunk 1 above)
Comment (TroyNeubauer) [reply src/main/include/CobConstants.h:3]: They are easily convertible to unique ints with static_cast<> [stackoverflow](https://stackoverflow.com/questions/8357240/how-to-automatically-convert-strongly-typed-enum-into-int)
Also we may not need to write our own serialization system if SetPersistent works as expected

File: src/main/cpp/Cob.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	RegisterKey(CobKey::ROBOT_POSITION_Y, "robot/position/y");
+	//...
+
+}
+
+void Cob::RegisterKey(CobKey key, std::string name, bool persistent) {
+	nt::NetworkTableEntry entry = s_Table->GetEntry(name);
+	s_Map[key] = entry;
+	if (persistent)
+		entry.SetPersistent();
```
Comment (TroyNeubauer) [inline src/main/cpp/Cob.cpp:28]: #7

File: src/main/include/subsystems/DriveTrain.h (hunk 2 above)
Comment (TroyNeubauer) [reply src/main/include/subsystems/DriveTrain.h:4]: Yeah

**Pull request merged**